### DISCUSSION TITLE
Add calendar triggers to Node-Red

### DISCUSSION THREAD

**Discussion by flyize** [zachowj/node-red-contrib-home-assistant-websocket#654] [Feature Request]
Would it be possible to get [calendar triggers](https://www.home-assistant.io/docs/automation/trigger/#calendar-trigger) added into Node-Red natively? Currently, NR can monitor the entity state of a calendar. However, that presents an issue when there are multiple events and one of them is an all-day event. The all-day event takes precedence and no triggers happen for any other events that day.

Native HA Automations can now use calendar triggers which gets around this issue. Would it be possible to get something like this added to NR?

For reference:
https://community.home-assistant.io/t/is-there-a-nr-equivalent-of-the-ha-automation-calendar-trigger/441700

**Comment by trooster** [comment]
Now the calendar in HA is getting more mature, this would be an excellent addition for calendar automations in Node-Red.

**Comment by jpravetz** [comment]
Are there updates from a contributor (someone that knows the code) regarding whether this is a possible feature, and how hard it would be to implement? Or have users discovered workarounds, or have users just settled on using Automations for this?
